docs/client-concepts/high-level/indexing/indexing.asciidoc

```diff
@@ -0,0 +1,186 @@
+:ref_current: https://www.elastic.co/guide/en/elasticsearch/reference/6.x
+
+:github: https://github.com/elastic/elasticsearch-net
+
+:nuget: https://www.nuget.org/packages
+
+////
+IMPORTANT NOTE
+==============
+This file has been generated from https://github.com/elastic/elasticsearch-net/tree/6.x/src/Tests/Tests/ClientConcepts/HighLevel/Indexing/Indexing.doc.cs. 
+If you wish to submit a PR for any spelling mistakes, typos or grammatical errors for this file,
+please modify the original csharp file found at the link and submit the PR with that change. Thanks!
+////
+
+[[indexing]]
+=== Indexing
+
+NEST has a number of ways in which documents can be indexed.
+
+==== Single Documents
+
+A single document can be indexed at a time, either synchronously or asynchronously
+
+[source,csharp]
+----
+var person = new Person
+{
+    Id = 1,
+    FirstName = "Martijn",
+    LastName = "Laarman"
+};
+
+var indexResponse = client.IndexDocument(person); <1>
+
+var asyncIndexResponse = await client.IndexDocumentAsync(person); <2>
+----
+<1> synchronous method that returns an `IIndexResponse`
+
+<2> asynchronous method that returns a `Task<IIndexResponse>` that can be awaited
+
+==== Multiple Documents with IndexMany
+
+Multiple documents can be indexed using the `IndexMany` and `IndexManyAsync` methods, again either synchronously or asynchronously.
+These methods are specific to the NEST client and wrap calls to the `_bulk` endpoint, providing a convenient shortcut to indexing
+multiple documents.
+
+[source,csharp]
+----
+var people = new []
+{
+    new Person
+    {
+        Id = 1,
+        FirstName = "Martijn",
+        LastName = "Laarman"
+    },
+    new Person
+    {
+        Id = 2,
+        FirstName = "Stuart",
+        LastName = "Cam"
+    },
+    new Person
+    {
+        Id = 3,
+        FirstName = "Russell",
+        LastName = "Cam"
+    }
+};
+
+var indexManyResponse = client.IndexMany(people); <1>
+
+if (indexManyResponse.Errors) <2>
+{
+    <3>
+    foreach (var itemWithError in indexManyResponse.ItemsWithErrors)
+    {
+        Console.WriteLine("Failed to index document {0}: {1}", itemWithError.Id, itemWithError.Error);
+    }
+}
+
+// Alternatively, documents can be indexed asynchronously
+var indexManyAsyncResponse = await client.IndexManyAsync(people); <4>
+----
+<1> synchronous method that returns an `IBulkResponse`
+
+<2> the response can be inspected for its success
```

Comment:
This is not _quite correct_; the bulk call can succeed and return a 200 response, but an individual bulk operation may have failed, and have an error. I think this should say something like

the response can be inspected to see if any of the bulk operations resulted in an error